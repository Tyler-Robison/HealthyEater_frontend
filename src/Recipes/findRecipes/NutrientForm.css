.NutrientForm-container {
    display: flex;
    flex-direction: column;
    align-items:center;
}

.NutrientForm-set-nutrients-btn {
    /* position: absolute; */
    /* left: 50%;
    transform: translate(-50%, -50%); */
    margin: 1rem 0;
}