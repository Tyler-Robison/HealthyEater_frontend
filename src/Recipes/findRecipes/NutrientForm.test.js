import React, { useState } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import NutrientForm from './NutrientForm';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext'

// using listState to set 
const DummyComponent = () => {
    const [fat, setFat] = useState(null)
    const [satFat, setSatFat] = useState(null)
    const [calories, setCalories] = useState(null)
    const [carbs, setCarbs] = useState(null)
    const [sugar, setSugar] = useState(null)
    const [sodium, setSodium] = useState(null)
    const [cholesterol, setCholesterol] = useState(null)
    const [protein, setProtein] = useState(null)

    return (
        <NutrientForm
            setFat={setFat}
            setSatFat={setSatFat}
            setSugar={setSugar}
            setSodium={setSodium}
            setCholesterol={setCholesterol}
            setCalories={setCalories}
            setCarbs={setCarbs}
            setProtein={setProtein}
        />
    )
}

it("renders without crashing", function () {
    <MemoryRouter>
        <ContextProvider >
            <DummyComponent />
        </ContextProvider>
    </MemoryRouter>
});


it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

test("can't input invalid values", async function () {

    render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );
    
    const fatInput = screen.getByLabelText('Max Fat (grams)')
    fireEvent.change(fatInput, { target: { value: -1 } });
    const setNutrientsBtn = screen.getByText('Set Nutrient Constraints')
    fireEvent.click(setNutrientsBtn)

    await waitFor(() => {
        expect(screen.getByText('Fat must be blank or positive number')).toBeInTheDocument();
    });
});