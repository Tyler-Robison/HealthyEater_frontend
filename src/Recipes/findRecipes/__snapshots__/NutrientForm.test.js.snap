// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="NutrientForm-container"
    >
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="fat"
          >
            Max Fat (grams)
          </label>
        </div>
        <input
          id="fat"
          name="maxFat"
          type="number"
          value=""
        />
      </div>
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="satFat"
          >
            Max Sat Fat (grams)
          </label>
        </div>
        <input
          id="satFat"
          name="maxSaturatedFat"
          type="number"
          value=""
        />
      </div>
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="carbs"
          >
            Max Carbs (grams)
          </label>
        </div>
        <input
          id="carbs"
          name="maxCarbs"
          type="number"
          value=""
        />
      </div>
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="calories"
          >
             Max Calories (kCals)
          </label>
        </div>
        <input
          id="calories"
          name="maxCalories"
          type="number"
          value=""
        />
      </div>
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="cholesterol"
          >
            Max Cholesterol (mg)
          </label>
        </div>
        <input
          id="cholesterol"
          name="maxCholesterol"
          type="number"
          value=""
        />
      </div>
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="sugar"
          >
            Max Sugar (grams)
          </label>
        </div>
        <input
          id="sugar"
          name="maxSugar"
          type="number"
          value=""
        />
      </div>
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="sodium"
          >
            Max Sodium (mg)
          </label>
        </div>
        <input
          id="sodium"
          name="maxSodium"
          type="number"
          value=""
        />
      </div>
      <div
        class="nutrient-input-div"
      >
        <div>
          <label
            for="protein"
          >
            Min Protein (grams)
          </label>
        </div>
        <input
          id="protein"
          name="minProtein"
          type="number"
          value=""
        />
      </div>
    </div>
    <div>
      <button
        class="general-btn NutrientForm-set-nutrients-btn"
        type="submit"
      >
        Update Values
      </button>
    </div>
  </form>
</DocumentFragment>
`;
